### PR TITLE
Add docs for summary type queries

### PR DESCRIPTION
**Pull request by rneatherway**

### CODE REVIEW

File: docs/query-metadata-style-guide.md [hunk 1]
```diff
@@ -157,6 +157,15 @@ If your query is a security query, use one or more `@tags` to associate it with
 
 When you tag a query like this, the associated CWE pages from [MITRE.org](http://cwe.mitre.org/index.html) will automatically appear in the reference section of its associated qhelp file.
 
+#### Metric/summary `@tags`
+
+Metrics and other pieces of summary information will be output in the SARIF format according to the [specification](https://docs.google.com/document/d/1V3zpkj0OGh8GEUVwACRx7fiafE5zklujAftZaYUyf9s/edit#heading=h.d6vfrrmrajbb) agreed with the Code Scanning team.
+
+Such queries should have the [`summary`](https://github.com/github/semmle-code/blob/4203403960b78c77c0a9563cce1bd85891cb4459/queryserver-client/src/com/semmle/api/QueryMetadata.java#L28-L32) tag.
```
Comment (intrigus-lgtm) [inline docs/query-metadata-style-guide.md:164]: This refers to a private repository.

File: docs/query-metadata-style-guide.md [hunk 2]
```diff
@@ -157,6 +157,15 @@ If your query is a security query, use one or more `@tags` to associate it with
 
 When you tag a query like this, the associated CWE pages from [MITRE.org](http://cwe.mitre.org/index.html) will automatically appear in the reference section of its associated qhelp file.
 
+#### Metric/summary `@tags`
+
+Metrics and other pieces of summary information will be output in the SARIF format according to the [specification](https://docs.google.com/document/d/1V3zpkj0OGh8GEUVwACRx7fiafE5zklujAftZaYUyf9s/edit#heading=h.d6vfrrmrajbb) agreed with the Code Scanning team.
```
Comment (intrigus-lgtm) [inline docs/query-metadata-style-guide.md:162]: This prompts me to login, so I guess you should probably replace this link with something that is "more" public.

File: docs/query-metadata-style-guide.md [hunk 3]
```diff
@@ -157,6 +157,12 @@ If your query is a security query, use one or more `@tags` to associate it with
 
 When you tag a query like this, the associated CWE pages from [MITRE.org](http://cwe.mitre.org/index.html) will automatically appear in the reference section of its associated qhelp file.
 
+#### Metric/summary `@tags`
+
+Queries with the `summary` tag represent metrics or other pieces of summary information. If SARIF output is used, the results of these queries can be found at `run[].properties.metricResults`.
```
Comment (adityasharad) [inline docs/query-metadata-style-guide.md:162]:
```suggestion
Queries with `@kind metric` and the `summary` tag represent metrics or other pieces of summary information. If SARIF output is used, the results of these queries can be found at `run[].properties.metricResults`.
```

File: docs/query-metadata-style-guide.md (hunk 2 above)
Comment (rneatherway) [reply docs/query-metadata-style-guide.md]: Thanks for the pointers. The internal links aren't suitable for inclusion here. Once `summary` queries are more widespread I hope we will provide external documentation for integration and then we could link to that here.

File: docs/query-metadata-style-guide.md (hunk 3 above)
Comment (adityasharad) [reply docs/query-metadata-style-guide.md:162]: Perhaps also add an entry to the `kind` and `tags` section above?

File: docs/query-metadata-style-guide.md [hunk 4]
```diff
@@ -157,6 +157,12 @@ If your query is a security query, use one or more `@tags` to associate it with
...
+
+Code Scanning may use tags to identify queries with specific meanings across languages. Currently, there is only one such tag: `lines-of-code`. The sum of the results for queries with this tag that return a single integer column ([example for JavaScript](https://github.com/github/codeql/blob/c47d680d65f09a851e41d4edad58ffa7486b5431/java/ql/src/Metrics/Summaries/LinesOfCode.ql)) is assumed by Code Scanning to be the lines of code under the source root present in the database. Each language should have exactly one query of this form.
+
```
Comment (adityasharad) [inline docs/query-metadata-style-guide.md:165]:
```suggestion
The `select` clause of a summary metric query must have one of the following result patterns:
- Just a `number`
  - This indicates a metric without a specific location in the codebase, for example the total lines of code in a codebase.
- A code `entity` followed by a `number`
  - This indicates a metric with a specific location in the codebase, for example the lines of code within a file. The `entity` here must have a valid location in the source code.
```

**Review by adityasharad — COMMENTED** [review]
Thanks! A few suggestions taken from the docs you originally linked.

File: docs/query-metadata-style-guide.md (hunk 3 above)
Comment (rneatherway) [reply docs/query-metadata-style-guide.md]: Just looking at this. Should metric queries _always_ have the `summary` tag?

Comment (adityasharad) [reply docs/query-metadata-style-guide.md]: More general metric queries without the `summary` tag are permitted by the engine, but not populated into `metricResults`. (My mental model is that "summary metric" queries are a subset of metric queries.)

Comment (rneatherway) [reply docs/query-metadata-style-guide.md]: Great. I've ended up moving your `select` suggestion under `QL area`, so the description is split across the doc as with alert queries. Perhaps if metric queries become more widespread we'll want two completely separate top level headings but this works for now IMO.

**Review by adityasharad — DISMISSED** [review]

**Comment by adityasharad** [comment]
Oh wait! Please retarget against `main`.

**Comment by rneatherway** [comment]
> Oh wait! Please retarget against `main`.

Done, thanks! Not sure how that happened.

**Review by adityasharad — APPROVED** [review]

**Pull request merged**